common/src/main/java/org/opensearch/ml/common/agui/AGUIInputConverter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.common.agui;
+
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_CONTENT;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_CONTEXT;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_FORWARDED_PROPS;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_MESSAGES;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_ROLE;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_RUN_ID;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_STATE;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_THREAD_ID;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_TOOLS;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_FIELD_TOOL_CALL_ID;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_CONTEXT;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_FORWARDED_PROPS;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_MESSAGES;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_RUN_ID;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_STATE;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_THREAD_ID;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_TOOLS;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_PARAM_TOOL_CALL_RESULTS;
+import static org.opensearch.ml.common.agui.AGUIConstants.AGUI_ROLE_USER;
+import static org.opensearch.ml.common.utils.StringUtils.getStringField;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.ml.common.FunctionName;
+import org.opensearch.ml.common.dataset.remote.RemoteInferenceInputDataSet;
+import org.opensearch.ml.common.input.execute.agent.AgentMLInput;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class AGUIInputConverter {
```

Comment:
I don't see any comprehensive input sanitization. This can be potential for injection attacks through malformed AG-UI JSON input